core/unit_test/view/TestViewCustomizationAllocationType.hpp

```diff
@@ -21,14 +21,36 @@
 namespace Foo {
 
 // ElementType which is actually just a tag type
-struct Bar {};
+struct Bar {
+  float vals[5];
+  KOKKOS_FUNCTION
+  float& operator[](size_t idx) { return vals[idx % 5]; }
+  KOKKOS_FUNCTION
+  const float& operator[](size_t idx) const { return vals[idx % 5]; }
+};
 
 // Reference type for Bar
+template <class T>
 struct BarRef {
-  double* ptr{nullptr};
+  T* ptr{nullptr};
   size_t size{0ul};
+
+  KOKKOS_FUNCTION
+  T& operator[](size_t idx) const { return ptr[idx]; }
+
+  // assignment from value type should work for deep_copy from scalar!
+  // Just will some defined values so we know we did it.
```

Comment:
I'm not sure what the last sentence means. What are the "defined values"?
```suggestion
  // For the assignment from value type to work for deep_copy from scalar,
  // we need this class to be convertible to Bar.
```